tests/drivers/redis/QueueTest.php

```diff
@@ -137,4 +138,41 @@ protected function tearDown()
         $this->getQueue()->redis->flushdb();
         parent::tearDown();
     }
+
+    public function testMoveExpired()
+    {
+        $job = $this->createSimpleJob();
+        $this->getQueue()->delay(1)->push($job);
+        //expect 1 msg should be recv
+        $msgCount = 0;
+        $this->getQueue()->messageHandler = function () use(&$msgCount) {
+            $msgCount++;
+        };
+
+        $fault = function () {
+            sleep(2);
+            $mockRedis = Instance::ensure([
+                'class' => RedisCrashMock::class,
+                'hostname' => getenv('REDIS_HOST') ?: 'localhost',
+                'database' => getenv('REDIS_DB') ?: 1,
+            ], 'yii\redis\Connection');
+
+            $queue = $this->getQueue();
+            $old = $queue->redis;
+            $queue->redis = $mockRedis;
+
+            try {
+                $queue->run(false);
+            }catch (\Exception $e){
+            }finally{
+                $queue->redis = $old;
+            }
+        };
+        $fault();
```

Comment:
Why do we need to wrap this code into function and execute it right away?

Reply:
I often use a function to wrap a clear and specific meaning. Originally, I wanted to use it here to describe a crash scenario.